apps/web/ee/pages/workflows/[workflow].tsx

```diff
@@ -0,0 +1,184 @@
+import { PencilIcon } from "@heroicons/react/solid";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { TimeUnit, WorkflowStep, WorkflowTriggerEvents } from "@prisma/client";
+import { isValidPhoneNumber } from "libphonenumber-js";
+import { useSession } from "next-auth/react";
+import { useRouter } from "next/router";
+import { useEffect, useState } from "react";
+import { useForm } from "react-hook-form";
+import { z } from "zod";
+
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import { Alert } from "@calcom/ui/Alert";
+import Loader from "@calcom/ui/Loader";
+import LicenseRequired from "@ee/components/LicenseRequired";
+import WorkflowDetailsPage from "@ee/components/workflows/WorkflowDetailsPage";
+import {
+  TIME_UNIT,
+  WORKFLOW_ACTIONS,
+  WORKFLOW_TEMPLATES,
+  WORKFLOW_TRIGGER_EVENTS,
+} from "@ee/lib/workflows/constants";
+
+import useMeQuery from "@lib/hooks/useMeQuery";
+import { trpc } from "@lib/trpc";
+
+import Shell from "@components/Shell";
+
+export type Option = {
+  value: string;
+  label: string;
+};
+
+export type FormValues = {
+  name: string;
+  activeOn: Option[];
+  steps: WorkflowStep[];
+  trigger: WorkflowTriggerEvents;
+  time?: number;
+  timeUnit?: TimeUnit;
+};
+
+function WorkflowPage() {
+  const { t } = useLocale();
+  const session = useSession();
+  const router = useRouter();
+  const me = useMeQuery();
+  const isFreeUser = me.data?.plan === "FREE";
+
+  const [editIcon, setEditIcon] = useState(true);
+  const [selectedEventTypes, setSelectedEventTypes] = useState<Option[]>([]);
+  const [isAllDataLoaded, setIsAllDataLoaded] = useState(false);
+
+  const formSchema = z.object({
+    name: z.string(),
+    activeOn: z.object({ value: z.string(), label: z.string() }).array(),
+    trigger: z.enum(WORKFLOW_TRIGGER_EVENTS),
+    time: z.number().gte(0).optional(),
+    timeUnit: z.enum(TIME_UNIT).optional(),
+    steps: z
+      .object({
+        id: z.number(),
+        stepNumber: z.number(),
+        action: z.enum(WORKFLOW_ACTIONS),
+        workflowId: z.number(),
+        reminderBody: z.string().optional().nullable(),
+        emailSubject: z.string().optional().nullable(),
+        template: z.enum(WORKFLOW_TEMPLATES),
+        sendTo: z
+          .string()
+          .refine((val) => isValidPhoneNumber(val))
+          .optional()
+          .nullable(),
+      })
+      .array(),
+  });
```

Comment:
This should live outside of React

Reply:
@zomars What exactly do you mean by that?

Reply:
It doesn't need to be inside the React component. This object will be re-created on every render which may have some unintended consequences. Also, since this schema doesn't depend on any state or props. We can safely put it outside the React component.